examples/default-values/schema.ts

```diff
@@ -27,15 +27,21 @@ export const lists = createSchema({
       assignedTo: relationship({
         ref: 'Person.tasks',
         many: false,
-        // Dynamic default: Find an anonymous user and assign the task to them
-        defaultValue: async ({ context }) => {
-          const anonymous = await context.lists.Person.findMany({
-            where: { name: { equals: 'Anonymous' } },
-          });
-          if (anonymous.length > 0) {
-            return { connect: { id: anonymous[0].id } };
-          }
-          // If we don't have an anonymous user return undefined so as not to apply any default
+        hooks: {
+          // Dynamic default: Find an anonymous user and assign the task to them
+          async resolveInput({ context, operation, resolvedData }) {
+            if (operation === 'create' && !resolvedData.assignedTo) {
+              const anonymous = await context.db.lists.Person.findMany({
+                where: { name: { equals: 'Anonymous' } },
+              });
+              if (anonymous.length > 0) {
+                return { connect: { id: anonymous[0].id } };
+              }
+            }
+            // If we don't have an anonymous user we return the value
+            // that was passed in(which might be nothing) so as not to apply any default
+            return resolvedData.assignedTo;
+          },
```

Comment:
This whole example will need to be updated but that can happen later when all the field have been updated